src/main/cluster.ts

```diff
@@ -172,29 +192,38 @@ export class Cluster implements ClusterModel {
   }
 
   @action
-  async refresh() {
+  async refresh(opts: ClusterRefreshOptions = {}) {
     logger.info(`[CLUSTER]: refresh`, this.getMeta());
     await this.whenInitialized;
     await this.refreshConnectionStatus();
     if (this.accessible) {
-      this.distribution = this.detectKubernetesDistribution(this.version)
-      const [features, isAdmin, nodesCount] = await Promise.all([
+      const [features, isAdmin] = await Promise.all([
         getFeatures(this),
         this.isClusterAdmin(),
-        this.getNodeCount(),
       ]);
       this.features = features;
       this.isAdmin = isAdmin;
-      this.nodes = nodesCount;
-      await Promise.all([
+      const refreshPromises = [
         this.refreshEvents(),
         this.refreshAllowedResources(),
-      ]);
-      this.ready = true
+      ]
+      if (opts.refreshMetadata) {
+        refreshPromises.push(this.refreshMetadata())
```

Comment:
Is this safe? What happens if collecting metadata takes time?

Reply:
Not waiting it anymore. Perhaps some additional logic could be added in future PRs.